src/core/server/saved_objects/migrationsv2/integration_tests/actions.test.ts

```diff
@@ -0,0 +1,544 @@
+  createIndex,
+  fetchIndices,
+  reindex,
+  searchForOutdatedDocuments,
+  SearchResponse,
+  setWriteBlock,
+  updateAliases,
+  waitForReindexTask,
+  ReindexResponse,
+  waitForPickupUpdatedMappingsTask,
+  pickupUpdatedMappings,
+  UpdateByQueryResponse,
+  updateAndPickupMappings,
+  UpdateAndPickupMappingsResponse,
+} from '../actions';
+import * as Either from 'fp-ts/lib/Either';
+import * as Option from 'fp-ts/lib/Option';
+
+const { startES } = kbnTestServer.createTestServers({
+  adjustTimeout: (t: number) => jest.setTimeout(t),
+});
+let esServer: kbnTestServer.TestElasticsearchUtils;
+
+describe('migration actions', () => {
+  let root: Root;
+  let start: InternalCoreStart;
+  let client: ElasticsearchClient;
+
+  beforeAll(async () => {
+    esServer = await startES();
+    root = kbnTestServer.createRootWithCorePlugins({
+      server: {
+        basePath: '/hello',
+      },
+    });
+
+    await root.setup();
+    start = await root.start();
+    client = start.elasticsearch.client.asInternalUser;
+
+    // Create test fixture data:
+    await createIndex(client, 'existing_index_1', { dynamic: true as any, properties: {} })();
+    const sourceDocs = ([
+      { _source: { title: 'doc 1' } },
+      { _source: { title: 'doc 2' } },
+      { _source: { title: 'doc 3' } },
+      { _source: { title: 'saved object 4' } },
+    ] as unknown) as SavedObjectsRawDoc[];
+    await bulkOverwriteTransformedDocuments(client, 'existing_index_1', sourceDocs)();
+
+    await createIndex(client, 'existing_index_2', { properties: {} })();
+    await createIndex(client, 'existing_index_with_write_block', { properties: {} })();
+    await bulkOverwriteTransformedDocuments(
+      client,
+      'existing_index_with_write_block',
+      sourceDocs
+    )();
+    await setWriteBlock(client, 'existing_index_with_write_block')();
+    await updateAliases(client, [
+      { add: { index: 'existing_index_2', alias: 'existing_index_2_alias' } },
+    ])();
+  });
+
+  afterAll(async () => {
+    await esServer.stop();
+    await root.shutdown();
+  });
+
+  describe('fetchIndices', () => {
+    it('resolves right empty record if no indices were found', async () => {
+      const task = fetchIndices(client, ['no_such_index']);
+      return expect(task()).resolves.toMatchInlineSnapshot(`
+                Object {
+                  "_tag": "Right",
+                  "right": Object {},
+                }
+              `);
+    });
+    it('resolves right record with found indices', async () => {
+      const res = (await fetchIndices(client, [
+        'no_such_index',
+        'existing_index_1',
+      ])()) as Either.Right<unknown>;
+
+      return expect(res.right).toEqual(
+        expect.objectContaining({
+          existing_index_1: {
+            aliases: {},
+            mappings: expect.anything(),
+            settings: expect.anything(),
+          },
+        })
+      );
+    });
+  });
+
+  describe('setWriteBlock', () => {
```

Comment:
Need to add tests for:
- When it succeeds it resolves right
- That subsequent write attempts to the index return an error